crypto/x509/by_dir.c

```diff
@@ -374,6 +374,11 @@ static int get_cert_by_subject(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
             ok = 1;
             ret->type = tmp->type;
             memcpy(&ret->data, &tmp->data, sizeof(ret->data));
+            /*
+             * Up the ref count of the contained object (note X509_OBJECT itself
+             * is not ref counted).
+             */
+            X509_OBJECT_up_ref_count(ret);
```

Comment:
[Documentation](https://www.openssl.org/docs/manmaster/man3/X509_LOOKUP_meth_set_get_by_subject.html) says this:

> Implementations should use either X509_OBJECT_set1_X509() or X509_OBJECT_set1_X509_CRL() to set the result.

This, lines 375-381 should be replaced with this:

``` C
            switch (tmp->type) {
            case X509_LU_X509:
                X509_OBJECT_set1_X509(ret, tmp);
                break;
            case X509_LU_X509_CRL:
                X509_OBJECT_set1_X509_CRL(ret, tmp);
                break;
            }
```

Reply:
Fixed. Fixup commit pushed.